tree.go

```diff
@@ -74,7 +73,7 @@ func (t *Tree) Insert(node *Node) error {
 	return nil
 }
 
-func (t Tree) find(name string) (int, *Node, error) {
+func (t Tree) binarySearch(name string) (int, *Node, error) {
```

Comment:
renamed this for clarity, since it took me a few minutes to realize why this is not super slow